lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -422,13 +422,17 @@ def tab_complete_stub(str, quote: nil)
       end
     }
 
-    # Verify that our search string is a valid regex
-    begin
-      Regexp.compile(str,Regexp::IGNORECASE)
-    rescue RegexpError
-      str = Regexp.escape(str)
+    # Convert into a case-insensitive regex
+    utf8_buf = str.dup.force_encoding('UTF-8')
+    if utf8_buf.valid_encoding?
+      Regexp.new(Regexp.escape(utf8_buf), Regexp::IGNORECASE)
+    else
+      # If the encoding is invalid, default to a regex that matches anything
+      //
```

Comment:
This will need an extra eye to understand what it was doing previously, and why it's no longer the same code 👀 

The encoding part is the essence of what I believe is needed to solve your original crash, but not the entire code block*

Reply:
Ok, I'll take another look into this.